Packages/Core/Editor/Generator/Templates/VariableTemplateFactory.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.Collections.ObjectModel;
+using UnityEditor;
+using UnityEngine;
+
+namespace UnityAtoms.Editor
+{
+    [InitializeOnLoad]
+    public static class VariableTemplateFactory
+    {
+        public const string atomName = "Variable";
+
+        static VariableTemplateFactory()
+        {
+            AtomGenerator.populators += AtomPopulator;
+            AtomGenerator.evaluators += AtomEvaluator;
+        }
+
+        private static void AtomPopulator(Type generatedType, System.Collections.Generic.Dictionary<string, Template> templates)
+        {
+            var atomTemplate = GetTemplate(generatedType);
+            templates.Add(atomName, atomTemplate);
+        }
+
+        private static void AtomEvaluator(Type generatedType, ReadOnlyDictionary<string, Generator.Solver> solvers)
+        {
+            var solver = solvers[atomName];
+
+            if (generatedType.IsEquatable())
+            {
+                solver.template.baseClass = typeof(EquatableAtomVariable<>).MakeGenericType(generatedType);
+
+                var body = string.Empty;
+
+                if (generatedType.IsColor())
+                {
+                    body =
+$@"
+        /// <summary>
+        /// Set Alpha of Color by value.
+        /// </summary>
+        /// <param name=""value"">New alpha value.</param>
+        public void SetAlpha(float value) => Value = new Color(Value.r, Value.g, Value.b, value);
+
+        /// <summary>
+        /// Set Alpha of Color by Variable value.
+        /// </summary>
+        /// <param name=""variable"">New alpha Variable value.</param>
+        public void SetAlpha(AtomBaseVariable<float> variable) => SetAlpha(variable.Value);
+    ";
+                }
+                else if (generatedType.IsNumeric())
+                {
+                    body =
+$@"
+        /// <summary>
+        /// Add value to Variable.
+        /// </summary>
+        /// <param name=""value"">Value to add.</param>
+        public void Add({generatedType.GenericName()} value) => Value += value;
```

Comment:
I believe we should use extension methods of the generic type instead of defining these in the generator. Eg `public static void Add(this AtomVariable<int> variable, int value) => variable.Value += value;`. That way we can keep the extension in the UnityAtoms core package and don't need to regenerate it every time and we avoid keeping the implementation as a string in this template.

Reply:
Disregard my comment above. Extension methods are not accessible from UnityEvents so it is not a good solution.

Reply:
Bummer, it was a nice dream.